pfio/v2/config.py

```diff
@@ -0,0 +1,50 @@
+import configparser
+import os
+from typing import Dict, Optional
+
+
+def _default_config_file():
+    path = os.getenv('PFIO_CONFIG_PATH')
+    if path:
+        return path
+
+    basedir = os.getenv('XDG_CONFIG_HOME')
+    if not basedir:
+        basedir = os.path.join(os.getenv('HOME'), ".config")
+
+    return os.path.join(basedir, "pfio.ini")
+
+
+def _reload_config():
+    global _config
+    config = configparser.ConfigParser()
+    configfile = _default_config_file()
+    config.read(configfile)
+    _config = config
+
+
+def add_custom_scheme(
+    name: str,
+    scheme: str,
+    data: Optional[Dict[str, str]] = None,
+) -> None:
+    if data is None:
+        data = {}
+    else:
+        data = data.copy()
+
+    data["scheme"] = scheme
+    _config[name] = data
+
+
+def get_custom_scheme(name: str) -> Optional[Dict[str, str]]:
+    try:
+        return dict(_config[name])
+    except KeyError:
+        return None
+
+
+_config = None
+
+
+_reload_config()
```

Comment:
I don't want to change the timing of loading configuration file. Currently it was read every time opening any `FS`, but with this line it'll be changed just once on global startup. How about keeping dynamically added dict separately, and merge it with configs from the file every time?

Reply:
Fixed for lazy loading: a016f14ff6f3f4cce13e7a6c1605718fbd6a16e3

But I just realized I didn't read your comment carefully.
Could you elaborate a bit more?
(Do you mean `get_custom_scheme()` would look up from both `_configs` and the separate dict? I don't get why the separate dict would be needed.)

Reply:
I thought that both the configuration file and the global dict `data` could be both updated anytime during the process lifetime. But maybe that'd be too much and ineffective.